htsohm/binning.py

```diff
@@ -16,10 +16,15 @@ def select_parents(run_id, children_per_generation, generation):
     is selected, a parent is randomly-selected from those materials within that bin.
     """
     # Each bin is counted...
-    bins_and_counts = session.query(func.count(Material.id), Material.methane_loading_bin,
-        Material.surface_area_bin, Material.void_fraction_bin).filter(
-        Material.run_id == run_id).group_by(Material.methane_loading_bin,
-        Material.surface_area_bin, Material.void_fraction_bin).all()[1:]
+    bins_and_counts = session \
+        .query(
+            func.count(Material.id), Material.methane_loading_bin, Material.surface_area_bin,
+            Material.void_fraction_bin
+        ) \
+        .filter(
+            Material.run_id == run_id).group_by(Material.methane_loading_bin,
+            Material.surface_area_bin, Material.void_fraction_bin
+        ).all()[1:]
```

Comment:
.group_by should be at the same level as .filter, so something like:

.filter(Material.run_id == run_id)
.group_by(
  Material.methane_loading_bin,
  Material.surface_area_bin, Material.void_fraction_bin
).all()[1:]